src/intersections/lineraysegmentplane.jl

```diff
@@ -0,0 +1,114 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    Return appropriate type for a geometry overlapping with a `Plane`
+
+### Notes
+
+- Ideally this would be a macro, but the geometry type isn't known at parse time, so it is implemented with multiple-dispatch instead
+"""
+function planeintersectionoverlapping(::Line)
+  return OverlappingLinePlane
+end
+
+function planeintersectionoverlapping(::Ray)
+  return OverlappingRayPlane
+end
+
+function planeintersectionoverlapping(::Segment)
+  return OverlappingSegmentPlane
+end
```

Comment:
```suggestion
_overlapping(::Line) = OverlappingLinePlane
_overlapping(::Ray) = OverlappingRayPlane
_overlapping(::Segment) = OverlappingSegmentPlane
```

Reply:
This is much tidier, I should have thought of this much quicker, thanks!